internal/errors/errors.go

```diff
@@ -263,10 +263,22 @@ func Join(errs ...error) error {
 	var e *joinError
 	switch x := errs[0].(type) {
 	case *joinError:
-		e = x
+		if x != nil {
+			e = x
+		} else {
+			e = &joinError{
```

Comment:
Since you are instantiating a new joinError in all cases, why not simply check if e is nil after the Switch and create a new one only if it is nil?

Reply:
Thank you for your feedback. I fixed it!
https://github.com/vdaas/vald/pull/2517/commits/8009a29734657e8fdfd41998cd7a983f5d128909